sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -845,11 +853,23 @@ public void setValueDeserializer(String valueDeserializer) {
       return toBuilder().setConsumerConfig(config).build();
     }
 
-    /** Returns a {@link PTransform} for PCollection of {@link KV}, dropping Kafka metatdata. */
+    /**
+     * Returns a {@link PTransform} for {@link PCollection} of {@link KV}, dropping Kafka metadata.
+     */
     public PTransform<PBegin, PCollection<KV<K, V>>> withoutMetadata() {
       return new TypedWithoutMetadata<>(this);
     }
 
+    /**
+     * Returns a {@link PTransform} for {@link PCollection} of {@link GenericRecord}, dropping Kafka
+     * keys and metadata. It has to be used only with {@link
+     * ConfluentSchemaRegistryDeserializerProvider}. See {@link KafkaIO} for more information on
+     * usage and configuration of reader.
+     */
+    public PTransform<PBegin, PCollection<GenericRecord>> withAvroSchemaValues() {
```

Comment:
We do not need an special PTransform for this. Notice that both SqlTransform and most Schema-based PTransforms do require PCollection<T> but with the specific condition that this PCollection should be encoded using a `SchemaCoder`. So we can just use the existing methods that produce `PCollection<K>` and `PCollection<V>`.